Automattic/wp-calypso
# Stats: Release Restricted Dashboard

Related to pejTkB-1uX-p2 pejTkB-1t3-p2

## Proposed Changes
Release the following paywall strategy changes for all site types.

|Segment | Strategy|
-- | --
Commercial sites | - No Paywall (No compulsory redirection to the interstitial)<br />- Users with 1k+ monthly views will see a restricted dashboard with only 7-day highlights + 7-day traffic in the chart and the rest of the modules all gated. <br />- Users with <1k monthly views will see a restricted dashboard with only advance module gated (the same dashboard as PWYW users) with a upgrade banner NOT dismissible.
Non-commercial sites | - No Paywall (No compulsory redirection to the interstitial)<br />- Users will see a restricted dashboard with only advance module gated and a dismissible upgrade banners.

cc @mikestottuk @jsnmoon @gibrown 

## Why are these changes being made?

* Improve conversion rate and reduce 1 star reviews
* Also:
  * Remove the paywall where users wouldn't be able to experience Stats
  * Gated modules individually with call for upgrade buttons
  * Unify the paywall experience between new users and existing users

## Testing Instructions

It's easier to test locally. Please test for Odyssey Stats and then test in Calypso with similar steps.

- Build Odyssey Stats and Jetpack locally
  - `jetpack build plugins/jetpack`
  - `STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev`
- Remove any Stats plans from your local test site
- Toggle the commercial status to non-commercial from the blog report card
- Ensure you see modules gated per the description above
- Toggle site to a commercial and tweak views to 1k+ here by adding `return 1001;` as first line of the function:
  - https://github.com/Automattic/wp-calypso/blob/f823d57fcdee7558b5dfbdaa9a400f73d4d0e40a/client/state/stats/plan-usage/selectors.ts#L7
- Open `/wp-admin/admin.php?page=stats`
- Ensure you see modules gated per the description above
- Purchase a commercial stats product
- Ensure the gated modules become available
- Remove the product
- Purchase Jetpack Complate
- Ensure the gated modules become available

### All Non-commercial sites

<img width="426" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/187ec51d-6194-46a4-ac32-d7572645cc23">

### Commercial sites with <1k monthly views (They would see a different upgrade banner than non-commercial sites)

<img width="424" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/3f7c213d-3df0-4183-9013-102d31cbeaa5">


### Commercial with 1k+ monthly views

|Traffic | Insights |
|--|--|
|<img width="423" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/cb6a4934-1f04-4b0b-90cc-988f08331e5c">| <img width="424" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/4c169fe6-60a7-48c5-8a78-2092199bc082">|



## Pre-merge Checklist
